tests/test_functions.py

```diff
@@ -207,6 +207,18 @@ def f4(g):
             assert answer == 54
             """)
 
+    def test_closure_vars_from_static_parent(self):
```

Comment:
@darius I like this test. I've added it in my fork here https://github.com/rocky/x-python/commit/e4782c0bdc15b2c3e8410623d588bd3d97a783d6

What I like about this kind of test is that it simplifies testing. Instead of comparing output in the test runner, all the test runner has to do is run the program and the execution of the program tests itself.

And that way you can simply run the interpeter without any test framework to see if the issue is resoved. Real simple.

Reply:
I've gotten around to studying why this test fails and it uncovers a fundamental flaw with how opcode arguments are passed in the current design, at least for "freeops" I think. Lacking a better place, I'll note it here. 

It may be a while before I fix it in x-python. 

The "deref" opcodes like `LOAD_DEREF`, also known as opcodes in the `hasfree[]` opcodes list,  are passed a string _name_, but in the [Python reference library](https://docs.python.org/3/library/dis.html?highlight=load_deref#opcode-LOAD_DEREF) of course, the operand is really an integer index. 

In this inerpreter, the frame cells are a dictionary whereas in CPython frame cells are an array. 

Normally, everything is fine because names are distinct. For example, you can't have two local variables called `a`.  But in the cells array, names don't have to be distinct.

In particular in this test, the variable `xs` appears in two scopes. So when a `LOAD_DEREF` does its lookup into a dictionary, it finds the wrong value since the dictonary can only have one key with value `xs`.